lib/actions/user.actions.ts

```diff
@@ -1,39 +1,69 @@
 import dbConnect from "@/database/dbConnect";
-import User from "@/database/models/user.model";
-import bcryptjs from "bcryptjs";
+import User, { IUser } from "@/database/models/user.model";
+import { revalidatePath } from "next/cache";
+import { UpdateUserParams } from "./shared.actions";
 
-interface CreateUserProps {
-  fullname: string;
-  username: string;
-  password: string;
-}
-
-export const getOneUser = async (username: string) => {
+export const getOneUser = async (email: string) => {
   try {
     await dbConnect();
-    const user = await User.findOne({ username })!;
+    const user = (await User.findOne({ email })) as IUser;
     return user;
   } catch (error) {
-    throw new Error("");
+    console.log(error);
+  }
+};
+export const getUserId = async (
+  email: string
+): Promise<string | null | undefined> => {
+  try {
+    await dbConnect();
+    const user = await User.findOne({ email });
+    return user?._id as string;
+  } catch (error) {
+    console.log(error);
   }
 };
-export const createNewUser = async ({
-  username,
-  fullname,
-  password,
-}: CreateUserProps) => {
+export const getUserById = async ({ _id }: { _id: string }) => {
+  try {
+    await dbConnect();
+    const user = await User.findById(_id);
+    return user as IUser;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export async function updateUser(params: UpdateUserParams) {
   try {
     await dbConnect();
-    const hashedPassword = await bcryptjs.hash(password, 5);
-    const newUser = User.create({
-      fullname,
-      username,
-      password: hashedPassword,
+
+    const { _id, updateData, path } = params;
+
+    await User.findOneAndUpdate({ _id }, updateData, {
+      new: true,
     });
 
-    // need to route to onboarding if
-    return newUser;
+    revalidatePath(path);
+  } catch (error) {
+    console.log(error);
+    throw error;
+  }
+}
+export const createNewUser = async (userData: Partial<IUser>) => {
+  try {
+    await dbConnect();
+    const user = await User.create(userData);
+    return user as IUser;
+  } catch (error) {
+    console.log(error);
+  }
+};
```

Comment:
Are we no longer hashing the password for a created user? We should never store a user's PW as plaintext or handle hashing outside of the server